core/services/ocr2/plugins/mercury/plugin.go

```diff
@@ -22,11 +29,36 @@ import (
 	mercuryv1 "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/mercury/v1"
 	mercuryv2 "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/mercury/v2"
 	mercuryv3 "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/mercury/v3"
+	"github.com/smartcontractkit/chainlink/v2/plugins"
 )
 
 type Config interface {
 	MaxSuccessfulRuns() uint64
 	ResultWriteQueueDepth() uint64
+	plugins.RegistrarConfig
+}
+
+// concrete implementation of MercuryConfig
+type mercuryConfig struct {
+	jobPipelineMaxSuccessfulRuns     uint64
+	jobPipelineResultWriteQueueDepth uint64
+	plugins.RegistrarConfig
+}
+
+func NewMercuryConfig(jobPipelineMaxSuccessfulRuns uint64, jobPipelineResultWriteQueueDepth uint64, pluginProcessCfg plugins.RegistrarConfig) Config {
+	return &mercuryConfig{
+		jobPipelineMaxSuccessfulRuns:     jobPipelineMaxSuccessfulRuns,
+		jobPipelineResultWriteQueueDepth: jobPipelineResultWriteQueueDepth,
+		RegistrarConfig:                  pluginProcessCfg,
+	}
+}
+
+func (m *mercuryConfig) MaxSuccessfulRuns() uint64 {
+	return m.jobPipelineMaxSuccessfulRuns
+}
+
+func (m *mercuryConfig) ResultWriteQueueDepth() uint64 {
+	return m.jobPipelineResultWriteQueueDepth
 }
 
 func NewServices(
```

Comment:
@KuphJr this `NewServices` func is the most relevant to you. you will not actually need this change in the near future. instead the point is that if you have Provider (eg https://github.com/smartcontractkit/chainlink-common/blob/main/pkg/types/provider_mercury.go) , that is chain agnostic (and derived from a relayer) the we can make a loop